### PR TITLE
 Unify datetime import in all files

### LINKED ISSUE #4577: Unifiy `datetime` import
The imports of the `datetime` module are all over the place. We currently have at least the following styles in our lib:

* `import datetime`
* `import datetime as dtm`
* `from datetime import datetime, timedelta`

Let's unify that to `import datetime as dtm`. I prefer this option since

1. Since there is `datetime.datetime`, a standalone `datetime` can be confusing. Using the toplevel module is preferable IMO.
2. `dtm` is shorter that `datetime`, plus it make `dtm.datetime` unambiguous`

### PR DESCRIPTION
**Pull request by cuevasrja**
# Unifiy datetime import in all files

@Jeam-zx
@Migueldsc12
@AnyaMarcanito
@henryg311
@cuevasrja

## Summary

Convert all datetime imports of the project to `import datetime as dtm`. Since there was no standard in importing functions from this library.

## Tests

It was verified that all changes passed the static check performed by the linters and other project tools with the `pre-commit run -a` command.

## Check-list

This checklist is a non-exhaustive reminder of things that should be done before a PR is merged, both for you as contributor and for the maintainers. Feel free to copy (parts of) the checklist to the PR description to remind you or the maintainers of open points or if you have questions on anything.

- [x] Added .. versionadded:: NEXT.VERSION, .. versionchanged:: NEXT.VERSION, .. deprecated:: NEXT.VERSION or .. versionremoved:: NEXT.VERSION to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes) 
- [x] Created new or adapted existing unit tests 
- [x] Documented code changes according to the [CSI standard](https://standards.mousepawmedia.com/en/stable/csi.html)
- [x] Added myself alphabetically to AUTHORS.rst (optional)
- [x] Added new classes & modules to the docs and all suitable __all__ s
- [x] Checked the [Stability Policy](https://docs.python-telegram-bot.org/stability_policy.html) in case of deprecations or changes to documented behavior

## Issue

Resolves #4577

### CODE REVIEW

File: telegram/constants.py
[the remark this answers is not in the document]
Comment (cuevasrja) [reply telegram/constants.py:1]: Thank you, but it continues failing in some Python Versions

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Thank you very much for the PR!
I just found two additional `import datetime` s in `test_jobqueue` and `test_businessmessagesdeletedhandler`, otherwise LGTM :)

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Thanks for the update! Now unfortunately something went wrong with merging master: Apparently you reverted the merge in https://github.com/python-telegram-bot/python-telegram-bot/pull/4605/commits/dda1e7ca1be439de94aac643457291ca3b7dad13 - please revert the revert 😅

**Review by Bibo-Joshi — APPROVED** [review]

**Comment by Bibo-Joshi** [comment]
Thank you very much for the contribution :)

**Pull request merged**